### PR TITLE
Add config to enable post-quantum secure TLS mode

### PR DESCRIPTION
**Pull request by hwupathum**
The following configs will be needed to enable PQC mode

```
[transport.https.openssl]
enabled = true
named_groups="x25519_kyber768:x25519"

[transport.https.sslHostConfig.properties]
protocols="TLSv1+TLSv1.1+TLSv1.2+TLSv1.3"
```

#### Related Issue
https://github.com/wso2/product-is/issues/20290

### CODE REVIEW

File: modules/distribution/src/bin/openssl-tls.sh [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+        BUILD_OPENSSL=true
+        echo "[Mode]: Building OpenSSL"
+    elif [ "$arg" = "--build_pqclib" ] || [ "$arg" = "-build_pqclib" ] || [ "$arg" = "build_pqclib" ]; then
+        BUILD_PQCLIB=true
+        echo "[Mode]: Building OQS Provider"
+    fi
+done
+
+# Check for build dependencies
+check_dependencies make cmake wget tar gcc
```
Comment (ashendes) [inline modules/distribution/src/bin/openssl-tls.sh:200]: Need to check for perl in red hat linux distros

File: modules/distribution/src/bin/openssl-tls.sh [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+        # For Intel (x86_64) architecture
+        OPENSSL_INSTALL_DIR=$(ls -d '/usr/local/Cellar/openssl@3/'*)
+    fi
+fi
+
+if [ $BUILD_OPENSSL = true ] && [ ! -d "$OPENSSL_INSTALL_DIR" ]; then
+    install_openssl "$TMP_DIR" "$OPENSSL_INSTALL_DIR"
+else
+    echo "OpenSSL is already installed"
+fi
```
Comment (ashendes) [inline modules/distribution/src/bin/openssl-tls.sh:266]: This block can be refactored into a parent `if [ $BUILD_OPENSSL = true ];` block.

File: modules/distribution/src/bin/openssl-tls.sh [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+    else
+        echo "OQS Provider is already installed"
+    fi
+fi
+
+# =============================== Install APR ====================================================
+
+if [ $BUILD_OPENSSL = true ]; then
+    rm -rf "$CARBON_HOME/native/apr"
+    install_apr "$TMP_DIR" "$CARBON_HOME/native/apr"
```
Comment (ashendes) [inline modules/distribution/src/bin/openssl-tls.sh:294]: Why is the APR installation forced here? Openssl and oqs installations above don't follow this approach.

File: modules/distribution/src/repository/resources/conf/templates/repository/conf/tomcat/catalina-server.xml.j2 [hunk 4]
```diff
@@ -20,8 +20,11 @@
 
 <Server port="8005" shutdown="SHUTDOWN" xmlns:svns="http://org.wso2.securevault/configuration">
 
-    <Service className="org.wso2.carbon.tomcat.ext.service.ExtendedStandardService" name="Catalina">
+    {% if transport.https.openssl.enabled is sameas true %}
+    <Listener className="org.apache.catalina.core.AprLifecycleListener" SSLEngine="on" />
+    {% endif %}
 
+    <Service className="org.wso2.carbon.tomcat.ext.service.ExtendedStandardService" name="Catalina">
```
Comment (ashendes) [inline modules/distribution/src/repository/resources/conf/templates/repository/conf/tomcat/catalina-server.xml.j2:27]: Is this relevant to the standard operation of IS?

File: modules/distribution/src/bin/openssl-tls.sh (hunk 3 above)
Comment (hwupathum) [reply modules/distribution/src/bin/openssl-tls.sh:294]: In APR, the absolute path is stored during installation. Therefore need to force installation just like Tomcat native

**Review by ashendes — APPROVED** [review]

**Review by ashendes — APPROVED** [review]
